fmriprep/workflows/bold/registration.py

```diff
@@ -327,35 +327,24 @@ def init_bold_t1_trans_wf(freesurfer, mem_gb, omp_nthreads, multiecho=False, use
     # Generate a reference on the target T1w space
     gen_final_ref = init_bold_reference_wf(omp_nthreads, pre_mask=True)
 
-    if not multiecho:
-        # Merge transforms placing the head motion correction last
-        nforms = 2 + int(use_fieldwarp)
-        merge_xforms = pe.Node(niu.Merge(nforms), name='merge_xforms',
-                               run_without_submitting=True, mem_gb=DEFAULT_MEMORY_MIN_GB)
-        if use_fieldwarp:
-            workflow.connect([
-                (inputnode, merge_xforms, [('fieldwarp', 'in2')])
-            ])
-
+    # Merge transforms placing the head motion correction last
+    nforms = 2 + int(use_fieldwarp)
+    merge_xforms = pe.Node(niu.Merge(nforms), name='merge_xforms',
+                           run_without_submitting=True, mem_gb=DEFAULT_MEMORY_MIN_GB)
+    if use_fieldwarp:
         workflow.connect([
-            # merge transforms
             (inputnode, merge_xforms, [
-                ('hmc_xforms', 'in%d' % nforms),
-                ('itk_bold_to_t1', 'in1')]),
-            (merge_xforms, bold_to_t1w_transform, [('out', 'transforms')]),
-            (inputnode, bold_to_t1w_transform, [('bold_split', 'input_image')]),
+                ('fieldwarp' if not multiecho else 'identity', 'in2')])
         ])
 
-    else:
-        from nipype.interfaces.fsl import Split as FSLSplit
-        bold_split = pe.Node(FSLSplit(dimension='t'), name='bold_split',
-                             mem_gb=DEFAULT_MEMORY_MIN_GB)
-
-        workflow.connect([
-            (inputnode, bold_split, [('bold_split', 'in_file')]),
-            (bold_split, bold_to_t1w_transform, [('out_files', 'input_image')]),
-            (inputnode, bold_to_t1w_transform, [('itk_bold_to_t1', 'transforms')]),
-        ])
+    workflow.connect([
+        # merge transforms
+        (inputnode, merge_xforms, [
+            ('hmc_xforms' if not multiecho else 'identity', 'in%d' % nforms),
+            ('itk_bold_to_t1', 'in1')]),
+        (merge_xforms, bold_to_t1w_transform, [('out', 'transforms')]),
+        (inputnode, bold_to_t1w_transform, [('bold_split', 'input_image')]),
+    ])
```

Comment:
I think what we want in general is the following

```Python
    merge_xforms = pe.Node(niu.Merge(3), name='merge_xforms',
                           run_without_submitting=True, mem_gb=DEFAULT_MEMORY_MIN_GB)
    workflow.connect([
        # merge transforms
        (inputnode, merge_xforms, [
            ('hmc_xforms' , 'in3'),
            ('fieldwarp', 'in2'),
            ('itk_bold_to_t1', 'in1')]),
        (merge_xforms, bold_to_t1w_transform, [('out', 'transforms')]),
        (inputnode, bold_to_t1w_transform, [('bold_split', 'input_image')]),
    ])

Reply:
We still want to vary the number of transforms depending on whether or not `use_fieldwarp` is active, no ?

Reply:
Oh, sorry, I didn't see this. No, the idea is that we make that input be `identity`.

Reply:
Opened: https://github.com/nipreps/sdcflows/pull/108